.github/workflows/release.yaml

```diff
@@ -0,0 +1,62 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(yarn -s run get-version)" = "${{ inputs.tag }}"
+      - name: Give user instruction
+        if: failure()
+        run: echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag ${{ inputs.tag }} must match the version $(yarn -s run get-version)"
+      # Setup .npmrc file to publish to GitHub Packages
+      - name: Configure node with GitHub Packages
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
```

Comment:
would it not be better to move this to `package.json`: 

```
"publishConfig":{"registry":"http://my-internal-registry.local"}
```

Reply:
Now that we know where we stand on registry yes, it would. I just wonder how it will behave when we do the `npm config set registry https://registry.npmjs.org/` and try to publish into npmjs, (I can test it, but if you know the answer I'll take it).

Reply:
Got my answer, if we do this `npm config set registry` will not work but using the flag `--registry` when running `npm publish` works to overwrite this setting.

Reply:
Actually broke the authentication for some reason when specifying GitHub's registry there, so I'll remove it.